.gitignore

```diff
@@ -1 +1,2 @@
 /target
+Cargo.lock
```

Comment:
Thanks for your kindly reply. I'm quite new to Rust and didn't notice the Cargo.lock file should also be kept, thanks for pointing out. 

The md5path is used to serve as a temporary file (as we don't know the title in the first place) and address the many-to-one mapping between the urls and pdf (i.e. a pdf identified by its title may come from different sources: arxiv, conference website etc.).  

By creating `url: Option<String>` we seem to assume one-to-one mapping between url and paper metadata. If we simply decide whether to download and create a new paper metadata by checking whether the url is contained by any metadata in `state.papers`, there will be a problem when a user try to download the same paper from two different urls. If this happens, the old pdf with the same title will be overwriten by the newly downloaded one, and we end up with two paper metadata for the same paper.

One possible workaround is to stop creating the new metadata and pdf file once we know that there exists one paper with the same title in `file_dir`, however, we won't be able to predict this as the second url for the same paper is never recorded. So we cannot prevent duplicate downloading of the second url of the same paper.

To perserve the many-to-one mapping, perhaps another way is to switch to `urls: Vec<String>`. This will require some additional handling of the collision case. Once we collided into the same pdf in `file_dir` from a second url of that paper, we may raise an error and let the `execute` function handle the duplications by searching `state.papers` given the existing title and add the second url to the metadata.  

Another not very elegant way is to make the symlink hidden files so that user won't be able to see them once they open the file browser.

What do you think?

Reply:
Sorry for being stubborn. It's just that I find the md5 symlinking approach too heavy for a simple problem (if I understood the problem correctly). I'd like to find an alternative if possible. It introduces two new crate dependencies just for duplication-checking, produces symlinks that have illegible names.

Reply:
Yes, I agree with you that symlink is too heavy. I haven't noticed that reason adds a suffix to the markdown. It is definitely a nice idea to handle the duplicated file. Treating different paper versions differently sounds more desired and easier to handle. 

I may not have time to work on this recently, please feel free to take the neccessray part of code and do the updates.

Reply:
Sure. Would you press the 'Allow edits from maintainers' button if you haven't already?

Reply:
Sure, it has been ticked already.